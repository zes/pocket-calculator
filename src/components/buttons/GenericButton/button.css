.button {
	background-color: #ffffff;
	border: 0.1em solid #292c33;
	border-radius: 0.3em;
	box-sizing: border-box;
	color: #292c33;
	font-weight: 600;
	height: 100%;
	transition: all 0.3s;
	width: 100%;
}

button:hover {
	background-color: #696969;
	color: #ffffff;
}

button:focus {
  outline: none;
}
