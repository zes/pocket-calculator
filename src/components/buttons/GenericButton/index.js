export { default } from './GenericButton.component';
