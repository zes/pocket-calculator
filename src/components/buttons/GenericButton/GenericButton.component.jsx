import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import './button.css';

export default class Button extends PureComponent {
  onClick = () => {
    const { action } = this.props;
    action();
  };

  render() {
    const { className, value } = this.props;
    const buttonClassName = `button ${className}`;
    return (
      <button type="button" className={buttonClassName} onClick={this.onClick}>
        {value}
      </button>
    );
  }
}

Button.propTypes = {
  action: PropTypes.func.isRequired,
  className: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
};
