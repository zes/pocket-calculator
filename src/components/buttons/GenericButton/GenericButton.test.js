import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import GenericButton from './GenericButton.component';

Enzyme.configure({ adapter: new Adapter() });

describe('[Component] <ClearButton />', () => {
  const props = {
    action: jest.fn(),
    className: 'clear',
    value: 'C,'
  };

  it('renders a snapshot', () => {
    const tree = renderer.create(<GenericButton {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call updateOperation when user click on button', () => {
    const resultButton = shallow(<GenericButton {...props} />);
    resultButton.instance().onClick();
    expect(props.action).toHaveBeenCalledWith();
  });
});
