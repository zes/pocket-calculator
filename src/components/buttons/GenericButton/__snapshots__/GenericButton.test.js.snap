// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] <ClearButton /> renders a snapshot 1`] = `
<button
  className="button clear"
  onClick={[Function]}
  type="button"
>
  C,
</button>
`;
